mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/definition/annotation/impl/AnnotationPojoTypeNodeMetadataContributorImpl.java

```diff
@@ -0,0 +1,278 @@
+			cleanedUpMaxDepth = null;
+		}
+
+		String[] includePathsArray = annotation.includePaths();
+		Set<String> cleanedUpIncludePaths;
+		if ( includePathsArray.length > 0 ) {
+			cleanedUpIncludePaths = new HashSet<>();
+			Collections.addAll( cleanedUpIncludePaths, includePathsArray );
+		}
+		else {
+			cleanedUpIncludePaths = null;
+		}
+
+		collector.indexedEmbedded( cleanedUpPrefix, annotation.storage(), cleanedUpMaxDepth, cleanedUpIncludePaths );
+	}
+
+	private <A extends Annotation> AnnotationMarkerBuilder<A> createMarkerBuilder(A annotation) {
+		MarkerMapping markerMapping = annotation.annotationType().getAnnotation( MarkerMapping.class );
+		MarkerMappingBuilderReference builderReferenceAnnotation = markerMapping.builder();
+		BeanReference builderReference =
+				toBeanReference(
+						builderReferenceAnnotation.name(),
+						builderReferenceAnnotation.type(),
+						MarkerMappingBuilderReference.UndefinedImplementationType.class
+				)
+						.orElseThrow( () -> log.missingBuilderReferenceInBridgeMapping( annotation.annotationType() ) );
+
+		// TODO check generic parameters of builder.getClass() somehow, maybe in a similar way to what we do in FunctionBridgeUtil
+		return beanResolver.resolve( builderReference, AnnotationMarkerBuilder.class );
+	}
+
+	private BridgeBuilder<? extends IdentifierBridge<?>> createIdentifierBridgeBuilder(
+			DocumentId annotation, PropertyModel<?> propertyModel) {
+		IdentifierBridgeBeanReference bridgeReferenceAnnotation = annotation.identifierBridge();
+		Optional<BeanReference> bridgeReference = toBeanReference(
+				bridgeReferenceAnnotation.name(),
+				bridgeReferenceAnnotation.type(),
+				IdentifierBridgeBeanReference.UndefinedImplementationType.class
+		);
+		IdentifierBridgeBuilderBeanReference builderReferenceAnnotation = annotation.identifierBridgeBuilder();
+		Optional<BeanReference> builderReference = toBeanReference(
+				builderReferenceAnnotation.name(),
+				builderReferenceAnnotation.type(),
+				IdentifierBridgeBuilderBeanReference.UndefinedImplementationType.class
+		);
+
+		if ( bridgeReference.isPresent() && builderReference.isPresent() ) {
+			throw log.invalidDocumentIdDefiningBothBridgeReferenceAndBridgeBuilderReference( propertyModel.getName() );
+		}
+		else if ( bridgeReference.isPresent() ) {
+			// The builder will return an object of some class T where T extends FunctionBridge<?, ?>, so this is safe
+			@SuppressWarnings( "unchecked" )
+			BridgeBuilder<? extends IdentifierBridge<?>> castedBuilder =
+					new BeanResolverBridgeBuilder( IdentifierBridge.class, bridgeReference.get() );
+			return castedBuilder;
+		}
+		else if ( builderReference.isPresent() ) {
+			// TODO check generic parameters of builder.getClass() somehow, maybe in a similar way to what we do in FunctionBridgeUtil
+			return beanResolver.resolve( builderReference.get(), BridgeBuilder.class );
+		}
+		else {
+			// The bridge will be auto-detected from the property type
+			return null;
```

Comment:
Wouldn't using an `Optional` make sense here? As `null` seems to be a valid use case we need to take care of.

Reply:
> As null seems to be a valid use case we need to take care of.

Yes it is.

But... this is a private method, and its return type will be used directly as an argument to `PojoPropertyNodeMappingCollector#identifierBridge`, which is also internal code.

If I did return an `Optional`, I would have to change `PojoPropertyNodeMappingCollector#identifierBridge` to accept an `Optional`. And then comes the controversy of "do not use `Optional`s as method parameters": https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
(By the way, IDEA displays a warning if you try to do that)

All that thinking seems a bit much just for internal code. Frankly I don't know where to stand here. Don't even know if it's worth discussing, especially for internal code. Maybe we should decide whether or not we want `Optional` parameters once for all? Some policy like "unless there's a good reason (performance-critical code, API readability), use `Optional` wherever a parameter is optional" or "never use `Optional` in method parameters".

Reply:
OK. I must admit that I don't make that much difference between internal code or not. I just try to have all the code following the same standard and be readable.

I'm a bit worried that if we sometimes use `Optional` and sometimes not, we will tend to forget about taking care of the `null` case. But I probably worry too much.

Let's do it this way for now.